models/classes/export/metadata/TestMetadataByClassExportHandler.php

```diff
@@ -77,4 +86,41 @@ public function export($formValues, $destPath)
 
         return $report;
     }
-}
+
+    protected function exportClass($formValues, $destPath): Report
+    {
+        $classToExport = $this->getClassToExport($formValues['classUri']);
+
+        $zip = new \ZipArchive();
+        $zipPath = $formValues['filename'] . '_' . time() . '.zip';
+        $zip->open($zipPath, \ZipArchive::CREATE);
+
+        /** @var TestExporter $exporterService */
+        $exporterService = $this->getServiceManager()->get(TestMetadataExporter::SERVICE_ID);
+
+        $tmpFiles = [];
+
+        foreach ($classToExport->getInstances(true) as $instance) {
+            $exporterService->setOption(
+                TestExporter::OPTION_FILE_NAME,
+                $instance->getLabel() . '_metadata_' . time() . '.csv'
+            );
+
+            $filePath = $exporterService->export($instance->getUri());
+            $zip->addFile($filePath, $instance->getLabel() . '.csv');
```

Comment:
As export is possible for any class, in the case when nested classes exist two or more tests with the same label, in the archive will be set only the last file. That might produce issues in runtime and increase the possibility of support requests. That's no good.

Reply:
Thanks! It makes sence. 
I've updated solution. Now it works fully recursive. This case won't be an issue anymore. Please have a look.